cell_locking.hh

```diff
@@ -229,14 +229,20 @@ private:
         static constexpr size_t compute_rehash_at_size(size_t bucket_count) {
             return bucket_count * max_load_factor::num / max_load_factor::den;
         }
+
+        // Try to rehash the set, if needed.
+        // The function may fail silently.
+        // Rehashing would be retried at a later time on failure.
         void maybe_rehash() {
             if (_cell_count >= _rehash_at_size) {
                 auto new_bucket_count = std::min(_cells.bucket_count() * 2, _cells.bucket_count() + 1024);
+                try {
                 auto buckets = std::make_unique<cells_type::bucket_type[]>(new_bucket_count);
 
                 _cells.rehash(cells_type::bucket_traits(buckets.get(), new_bucket_count));
                 _buckets = std::move(buckets);
-
+                } catch (...) {
+                }
                 _rehash_at_size = compute_rehash_at_size(new_bucket_count);
```

Comment:
Please add a comment here, explaining the logic.

I can see someone reading this and saying it's an obvious bug, since new_bucket_count doesn't match _buckets.size().

Reply:
Also, add a warning, otherwise we won't know that this path happens in practice.